### ISSUE TITLE
Formatting date (yyyy-mm-dd) subtracts a day

### ISSUE THREAD

**Issue by holmok** [felixge/node-dateformat#108]
Here's how to reproduce it:
```javascript
> const DateFormat = require('dateformat');
undefined
> const date = new Date('2020-01-01')
undefined
> date
2020-01-01T00:00:00.000Z
> DateFormat(date,'yyyy-mm-dd')
'2019-12-31'
```

using version :   "dateformat": "^3.0.3",

**Comment by jamiesauve** [comment]
This is still an issue after upgrading to dateformat 4.4.1

**Comment by jamiesauve** [comment]
After doing some research, I don't think this is actually an issue, just some non-intuitive behavior on the part of Javascript's native `Date` constructor. I've noticed that I get the same behavior if I try `new Date("2021-01-11")` in the console, but if I type `new Date("2021-01-11T00:00:00")` it works fine. This library responds the same way (presumably because it's based on JS's Date). 
More knowledgeable folks than I have explained it here: https://stackoverflow.com/questions/7556591/is-the-javascript-date-object-always-one-day-off

**Comment by Giszmo** [comment]
Sorry but why would it not be an issue? If the underlying Date class is to blame, then don't use it but

```
> dateFormat('2021-03-10', 'isoDate')
'2021-03-09'
```

is painful to look at. As a user who wants to format a string that happens to already be in isoDate format to isoDate format should not have to deal with the date changing in the process.

In my project, setting the timezone to UTC:

```
process.env.TZ = 'UTC' // this has to be run before any use of Dates.
```

works but of course that's a terrible hack which removes all local timezone handling.

**Comment by maxima120** [comment]
@Giszmo I agree with Jamie. the JS is terrible in so many ways.. I dont know why there is no Linux-like revolution in web-dev. It seems ridiculous to use 1995 technology in 2021.. Dates handling is right laugh.. Having said that the browsers dont have correct way to deal with locale so date in JS cant be "fixed" the way you want (I want that too, but the problem is beyond this package reach).. So in a sense you are right too - dont use JS date but then you already have packages which do just that, only their size in prod is in MBs instead of 20Kb like this one.. If you are happy with that - pick another package.
